src/Actions/Product/UpdateProduct.php

```diff
@@ -85,6 +90,30 @@ public function performAction(): Model
                 );
         }
 
+        if ($productCrossSellings !== null) {
+            $activeProductCrossSellings = [];
+            foreach ($productCrossSellings as $productCrossSelling) {
+                $productCrossSelling['product_id'] = $product->id;
+                if ($productCrossSelling['id'] ?? false) {
+                    $action = UpdateProductCrossSelling::make($productCrossSelling);
+                } else {
+                    $action = CreateProductCrossSelling::make($productCrossSelling);
+                }
+
+                $activeProductCrossSellings[] = $action->checkPermission()->validate()->execute()->id;
```

Comment:
same as always, rethink if its that smart if an exception is thrown at this point in the code.